src/coreclr/CMakeLists.txt

```diff
@@ -206,15 +210,21 @@ if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSI
 endif()
 
 add_subdirectory(utilcode)
-add_subdirectory(gcinfo)
-add_subdirectory(jit)
 add_subdirectory(inc)
 
 if(CLR_CMAKE_HOST_UNIX)
     add_subdirectory(palrt)
 endif(CLR_CMAKE_HOST_UNIX)
 
-add_subdirectory(vm)
+if(CLR_CMAKE_BUILD_SUBSET_JIT)
+  add_subdirectory(gcinfo)
+  add_subdirectory(jit)
+  add_subdirectory(vm)
```

Comment:
Hmm, `CLR_CMAKE_BUILD_SUBSET_JIT` is defined but it does not seem to be actually used today. And the way you are using does not match its name, e.g. we do not consider "vm" to be a JIT. It may be best to delete it.

I am wondering whether the existing `CLR_CMAKE_BUILD_SUBSET_RUNTIME` is close what is needed here.

Reply:
Well, the `build-runtime` script currently builds quite a number of artifacts, in particular including:

- the IL tools (ilasm, ildasm)
- the "JIT" (if that's the correct term ...), including libclrjit.so and variants, libjitinterface.so, and corerun
- the "runtime", including libcoreclr.so, libmscordacore.so, libmscordbi.so, createdump, crossgen, libsuperpmi ...

There is an option `-skipruntime` which omits building the artifacts I listed under "runtime" above, and still builds those listed under "JIT" above as well as the IL tools.

There is also an option `-skipjit`, which currently does nothing.  What my patch does is to change this option to skip building the artifacts listed under "JIT" above, so that if you use both `-skipjit` and `-skipruntime` you only get the IL tools.

I guess I could instead change `-skipruntime` as you suggest to only build the IL tools.  But then there would be no way of getting back the current `-skipruntime` behavior of building just the "JIT" artifacts.

I don't really know whether there is any particular use case for this behavior, but I didn't want to simply remove it.  I'd be happy to implement whatever you think would be the best option -- but I'll need some guidance here.

Reply:
I think we should:
- Add `CLR_CMAKE_BUILD_SUBSET_ILTOOLS`
- Delete `skipjit` option since it does nothing
- Add `skipiltools` option that allows skipping iltools

Reply:
The problem is that on the (Mono-only) IBM Z target we are currently developing I cannot build the `vm` or `jit` directories, because they are simply not implemented for the platform.  And right now, every way to invoke `build-runtimes` **does** attempt to build those two directories.  So the main purpose of this patch is to add an option that will **skip** those directories so I can build just the IL tools (which do work).

That's why I used the existing `skipjit` to implement this.  If this deleted, then I still need _some_  way to skip those directories.

Reply:
`-skipalljits -skipruntime` would build just the iltools.

I do not really like specifying what to build by skipping parts that should not be built. I think it would be better to specify subset to build, similar to how they are specified in the top level script; but that would be more significant change.

Reply:
> `-skipalljits -skipruntime` would build just the iltools.

Well, it doesn't now.  `-skipalljits` currently skips building the JITs targeting other platforms as cross-compiler, but it still builds the native JIT (and always builds `vm`).   So this doesn't work for what I need either.   I guess this could be changed instead to build no JIT at all, but then there would again be no way to get the current behavior of `-skipalljits` back.

 > I do not really like specifying what to build by skipping parts that should not be built. I think it would be better to specify subset to build, similar to how they are specified in the top level script; but that would be more significant change.

Agreed.

Reply:
`skipruntime` can also skip the native JIT.

Reply:
Ok, that would work for me, and simplify things a bit.

In removing `-skipjit`, should I also remove the places that invoke it (e.g. the `ClrJitSubset` property and the `clr.jit` subset)?

Reply:
Yes

Reply:
Ok, will do.  Thanks!